lib/api/api_auth_webauthn.go

```diff
@@ -126,55 +120,33 @@ func (s *webauthnService) storeState(state config.WebauthnState) error {
 	return s.miscDB.PutBytes(s.miscDBKey, stateBytes)
 }
 
-func (s *webauthnService) WebAuthnID() []byte {
-	return s.userHandle
+func (s *webauthnService) user(guiCfg config.GUIConfiguration) webauthnLibUser {
+	return webauthnLibUser{
+		service: s,
+		guiCfg:  guiCfg,
+	}
 }
 
-func (s *webauthnService) WebAuthnName() string {
-	return s.cfg.GUI().User
+type webauthnLibUser struct {
+	service *webauthnService
+	guiCfg  config.GUIConfiguration
```

Comment:
The only field you are using is `guiCfg.User`, so you might as well add/pass only that instead of the entire config.

Reply:
Not quite - the `webauthnLib.User.WebAuthnCredentials()` method implementation calls `u.service.EligibleWebAuthnCredentials(u.guiCfg)` which references `guiCfg.WebauthnRpId`, so at least those two are both needed.

Reply:
Oh right, thanks for pointing that out.